simplesat/constraints/constraint_modifiers.py

```diff
@@ -96,6 +96,21 @@ def update(self, other_modifiers):
         self.allow_newer.update(other_modifiers.allow_newer)
         self.allow_older.update(other_modifiers.allow_older)
 
+    def remove(self, packages):
+        """ Remove all modifiers related to `packages`.
+
+        Parameters
+        ----------
+        packages : an iterable of strings
+            The package names that should be completely removed.
+        """
+        disallowed = (type(b""), type(u""))
+        if isinstance(packages, disallowed):
```

Comment:
We check for strings because it's easy to accidentally pass them instead of collections, which is bad.

`constraint.remove(('foo'))` would not do what you want.